src/nvim/buffer_defs.h

```diff
@@ -808,6 +808,7 @@ struct file_buffer {
 
   MarkTree b_marktree[1];
   Map(uint32_t, uint32_t) b_extmark_ns[1];         // extmark namespaces
+  size_t b_virt_text;           // number of virtual texts
```

Comment:
this should changed to `b_virt_text_inline` (and only count extmarks with inline virtual text)

Reply:
Done! 

For now I just added `&& decor->virt_text_pos == kVTInline` to the checks, but maybe it's worth introducing a helper function
 ```
bool decor_has_inline_virtual_text(Decoration *decor) { 
  return kv_size(decor->virt_text) && decor->virt_text_pos == kVTInline; 
}
```
for this?